parsl/data_provider/files.py

```diff
@@ -46,9 +46,13 @@ def __init__(self, url, dman=None, cache=False, caching_dir="."):
         self.scheme = parsed_url.scheme if parsed_url.scheme else 'file'
         self.netloc = parsed_url.netloc
         self.path = parsed_url.path
+        self.registered = False
```

Comment:
I'm pretty uncomfortable with `File` being modified to have what is pretty much executor-specific behaviour.

However I think we don't have a better abstraction at the moment so I can't really give any constructive direction here.  I'll try to formulate a more helpful opinion.